File: src/Other/Internals/Menu/UIElementHelpers.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace OpenSilver.Samples.Showcase
{
    public static class UIElementHelpers
    {
        public static Task WaitForLoadedAsync(FrameworkElement element)
        {
            if (element.IsLoaded)
            {
                return Task.CompletedTask;
            }

            var tcs = new TaskCompletionSource<object>();
            RoutedEventHandler handler = null;

            handler = (s, e) =>
            {
                element.Loaded -= handler;
                tcs.SetResult(null);
            };

            element.Loaded += handler;
            return tcs.Task;
        }

    }
}

File: src/Other/Internals/Menu/UIElementHelpers.fs
namespace OpenSilver.Samples.Showcase

open System.Threading.Tasks
open System.Windows

type UIElementHelpers() =
    static member WaitForLoadedAsync(element: FrameworkElement) : Task =
        if element.IsLoaded then
            Task.CompletedTask
        else
            let tcs = new TaskCompletionSource<obj>()
            let mutable handler : RoutedEventHandler = null
            
            let handler = RoutedEventHandler(fun s e ->
                element.Loaded.RemoveHandler(handler)
                tcs.SetResult(null)
            )
            
            element.Loaded.AddHandler(handler)
            tcs.Task
